### PR TITLE
NCTL: nctl-exec-upgrade-scenario-3

### PR DESCRIPTION
**Pull request by TomVasile**
Changes: 
- `nctl-exec-upgrade-scenario-3` 
    - Added to `activate`
    - Added to `staging|trying` bors
- `upgrade_scenario_3.accounts.toml` to `$NCTL/sh/scenarios/accounts_toml` for use in scenario
- `upgrade_scenario_3.chainspec.toml.in ` to `$NCTL/sh/scenarios/chainspecs` for use in scenario
- `setup_from_stage.sh` modified for custom chainspecs
- `await_n_eras` modified for variable sleep interval

Issue: https://app.zenhub.com/workspaces/cn-sre-60d363546afb85000e491ff6/issues/casper-network/sre/9

### CODE REVIEW

File: utils/nctl/sh/scenarios-upgrades/upgrade_scenario_03.sh [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+# ----------------------------------------------------------------
+# Synopsis.
+# ----------------------------------------------------------------
+
+# 1. Start v1 running at current mainnet commit.
+# 2. Execute some deploys to populate global state a little
+# 3. Upgrade all running nodes to v2
+# 4. Assert v2 nodes run & the chain advances (new blocks are generated)
+# 5. Start passive nodes.  The launcher should cause the v1 node to run, it should exit and the v2 node should then catch up.
+# 6. Assert passive nodes are now running.
```
Comment (goral09) [inline utils/nctl/sh/scenarios-upgrades/upgrade_scenario_03.sh:11]: The original ticket mentions bonding/bidding/auction actions across upgrade points but this description seems to be outdated. Can you update it to match the test?

Comment (TomVasile) [reply utils/nctl/sh/scenarios-upgrades/upgrade_scenario_03.sh]: Ninja emoji here

**Comment by TomVasile** [comment]
bors try

**Comment by TomVasile** [comment]
bors try

**Review by sacherjj — APPROVED** [review]
This looks good and tests what we are wanting.

**Comment by goral09** [comment]
How does this determine which release branch start from @TomVasile ? I.e. now the released version is `1.2.1` and I would imagine that it starts with that and upgrades to `dev`. Will it also automatically know to start from `1.3.0` when that's released? etc.

**Comment by TomVasile** [comment]
> How does this determine which release branch start from @TomVasile ? I.e. now the released version is `1.2.1` and I would imagine that it starts with that and upgrades to `dev`. Will it also automatically know to start from `1.3.0` when that's released? etc.

Thats taken care of here: https://github.com/casper-network/casper-node/blob/dev/ci/nctl_upgrade.sh#L84-L114

**Comment by TomVasile** [comment]
bors try

**Comment by TomVasile** [comment]
bors try

**Pull request merged**